src/main/webapp/dashboard.test.js

```diff
@@ -0,0 +1,132 @@
+import * as dashboard from './dashboard.js';
+import '@testing-library/jest-dom/extend-expect';
+import {screen} from '@testing-library/dom';
+
+/**
+ * @fileOverview this file is essentially testing that if correct information
+ * about a user is gotten from the UserSevlet, It will be displayed in some way
+ * on the Dashboard.html page.
+ */
+global.mailtouiApp = {run: () => {}};
+
+const mockUserAllFields = {
+  interestTags: ['interest1', 'interest2'],
+  projects: [
+    {repositoryId: 'mockRepoID1', name: 'mockProjectName1'},
+    {repositoryId: 'mockRepoID2', name: 'mockProjectName2'},
+  ],
+  bio: 'This is the mock bio.',
+  email: 'samialves@google.com',
+  gitHubID: 'mockGitID',
+  gitHubURL: 'http://localhost/mockGitID',
+  image: 'http://localhost/image.jpg',
+  location: 'Mockville, Mocktana',
+  name: 'Mocky the Mockster',
+};
+
+const mockUserMissingOptionalFields = {
+  name: 'Mocky the Mockster',
+  image: 'http://localhost/image.jpg',
+  gitHubID: 'mockGitID',
+  gitHubURL: 'http://localhost/mockGitID',
+};
+
+// Note: Update this when the dashboard gets changed
+/**
+ * Function to replicate the DOM of Dashboard.html
+ */
+function resetDocumentBody() {
+  document.body.innerHTML = `
+  <div id="about-me" class="container tab-pane active"><br>
+    <div class="card card-holder col-12 ">
+      <div id="about-me-card-body" class="card-body">
+        <img id="user-image" class="card-img-top img-thumbnail p-2 m-2 
```

Comment:
```suggestion
        <img id="${dashboard.USER_IMAGE_ID}" class="card-img-top img-thumbnail p-2 m-2 
```
Javascript template strings have interpolation, might as well use the exported IDs here as well.